### PR TITLE
Misc. typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3`

### CODE REVIEW

File: IPython/core/completer.py [hunk 1]
```diff
@@ -949,7 +949,7 @@ def _formatparamchildren(parameter) -> str:
 
     Jedi does not expose a simple way to get `param=value` from its API.
 
-    Prameter
+    Parameter
     ========
```
Comment (takluyver) [inline IPython/core/completer.py]: Can you extend the underline here? I think it needs to be the same length as the heading.

**Review by takluyver — COMMENTED** [review]
Thanks!

**Comment by luzpaz** [comment]
Revised and rebased

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**